src/osd/osd_internal_types.h

```diff
@@ -434,21 +434,21 @@ class ObcLockManager {
   }
 
   void put_locks(
-    list<pair<hobject_t, list<OpRequestRef> > > *to_requeue,
+    list<pair<ObjectContextRef, list<OpRequestRef> > > *to_requeue,
     bool *requeue_recovery,
     bool *requeue_snaptrimmer) {
-    for (auto p: locks) {
+    for (auto& p: locks) {
```

Comment:
Just this line is probably a big part of the improvement?

Reply:
Exactly.